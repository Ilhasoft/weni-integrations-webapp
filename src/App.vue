<template>
  <div>
    <Carousel />
    <NavBar />
    <router-view />
  </div>
</template>

<script>
  import NavBar from './components/NavBar.vue';
  import Carousel from './components/Carousel.vue';
  import '@weni/unnnic-system';

  export default {
    name: 'App',
    components: {
      NavBar,
      Carousel,
    },
  };
</script>

<style lang="scss">
  @import '~@weni/unnnic-system/dist/unnnic.css';
  @import '~@weni/unnnic-system/src/assets/scss/unnnic.scss';
  body {
    margin: $unnnic-spacing-inset-md;
    background-color: $unnnic-color-background-snow;
    font-family: $unnnic-font-family-secondary, Avenir, Helvetica, Arial, sans-serif;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    box-sizing: border-box;
  }
</style>
