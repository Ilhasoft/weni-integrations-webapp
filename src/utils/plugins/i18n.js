import Vue from 'vue';
import VueI18n from 'vue-i18n';

Vue.use(VueI18n);

import en from '../../locales/en.json';
import pt_br from '../../locales/pt_br.json';
import es_es from '../../locales/es_es.json';

const languages = {
  'en-us': en,
  'pt-br': pt_br,
  es: es_es,
};

const messages = Object.assign(languages);

const i18n = new VueI18n({
  locale: 'en-us',
  fallbackLocale: 'en-us',
  messages,
});

export default i18n;
