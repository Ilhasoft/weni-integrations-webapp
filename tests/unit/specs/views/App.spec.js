jest.mock('@/api/auth', () => {
  return {
    getFlowToken: jest.fn(),
  };
});

import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import App from '@/App.vue';
import i18n from '@/utils/plugins/i18n';
import VueRouter from 'vue-router';
const router = new VueRouter();

const localVue = createLocalVue();
localVue.use(VueRouter);
localVue.use(Vuex);

describe('App.vue', () => {
  let wrapper;
  let actions, store;

  beforeEach(() => {
    actions = {
      getFlowToken: jest.fn(),
    };

    store = new Vuex.Store({
      actions,
    });

    wrapper = shallowMount(App, {
      localVue,
      i18n,
      store,
      router,
    });
  });

  it('should be rendered properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
