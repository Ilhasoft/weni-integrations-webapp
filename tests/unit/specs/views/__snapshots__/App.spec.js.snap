// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue should be rendered properly 1`] = `
<div class="app">
  <router-view-stub name="default" class="content"></router-view-stub>
  <div class="survey-container">
    <survey-stub></survey-stub>
  </div>
</div>
`;
